airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/page_increment.py

```diff
@@ -35,7 +43,7 @@ def initial_token(self) -> Optional[Any]:
 
     def next_page_token(self, response: requests.Response, last_records: List[Mapping[str, Any]]) -> Optional[Any]:
         # Stop paginating when there are fewer records than the page size or the current page has no records
-        if (self.page_size and len(last_records) < self.page_size) or len(last_records) == 0:
+        if (self._page_size and len(last_records) < self._page_size.eval(self.config)) or len(last_records) == 0:
```

Comment:
the page size only has context on the config, so it could be evaluated in the `__post_init__`. The benefit is it would fail at startup instead of halfway through the sync

Reply:
Updated.